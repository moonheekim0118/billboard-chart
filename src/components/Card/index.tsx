import React from 'react';
import styled from 'styled-components';

const Card = () => {
    return (
        <Container>
            {/* <MusicInfo>
                <Rank>{props.music.rank}</Rank>
                <Info>
                    <Title>{props.music.title}</Title>
                    <Artist>{props.music.artist}</Artist>
                </Info>
            </MusicInfo>
            <LastWeek>으잉</LastWeek> */}
        </Container>
    );
};

const Container = styled.div`
    display: flex;
    justify-content: space-between;
    align-item: center;
    width: 100%;
    height: 80px;
    background-color: #fff;

    margin-top: 25px;
    box-shadow: 0px 0px 2px 0px rgba(255, 255, 255, 1);
    border-radius: 10px;

    cursor: pointer;
    transition: box-shadow 0.5s ease;
    &:hover {
        box-shadow: 0px 0px 8px 0px rgba(255, 255, 255, 1);
    }
`;

const MusicInfo = styled.div`
    display: flex;
`;

const Rank = styled.span`
    font-size: 1.2em;
    margin-right: 35px;
    border-right: 1px solid #eaeae1;
    padding: 25px 20px;

    color: #34495e;
`;

const Info = styled.div`
    display: flex;
    flex-direction: column;
    padding: 15px 20px;
`;

const Title = styled.span`
    font-size: 1.2em;
`;

const Artist = styled.span`
    font-size: 1.1em;
`;

const LastWeek = styled.div`
    padding: 25px 20px;
`;
export default Card;
