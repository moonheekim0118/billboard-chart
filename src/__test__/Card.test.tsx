import 'jsdom-global/register';
import React from 'react';
import Card from '../components/Card';
import { HashRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('<Card/>', () => {
    let container = null;
    const mockData = {
        api_path: 'sample',
        id: 1,
        title: 'sample',
        song_art_image_thumbnail_url: 'sample',
        primary_artist: {
            id: 1,
            name: 'sample',
        },
    };

    it('renders correctly', () => {
        container = mount(
            <HashRouter>
                <Card data={mockData} search={true} />
            </HashRouter>
        );
    });

    it('matches snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});
