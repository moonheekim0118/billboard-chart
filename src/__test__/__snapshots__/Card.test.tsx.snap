// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> matches snapshot 1`] = `"<div class=\\"sc-bdfBwQ iqxdzn\\"><div class=\\"sc-gsTCUz kxzozt\\"><img alt=\\"album art\\" height=\\"150px\\" width=\\"150px\\" src=\\"sample\\" class=\\"sc-dlfnbm jDravJ\\"><div class=\\"sc-hKgILt eIKxVQ\\"><span class=\\"sc-eCssSg fgXKYL\\">sample</span><a class=\\"sc-jSgupP eSwkBk\\" href=\\"#/artist/1\\">sample</a></div></div></div>"`;
